HIP/hip-1084.md

```diff
@@ -0,0 +1,118 @@
+ - `EOA`: The account that created and signed the `evm tx` and submitted it to a `relay`. This is the `from` in the
+          [Ethereum Transaction Schema](https://github.com/ethereum/execution-apis/blob/main/src/eth/transaction.yaml)
+          often noted as (tx.from)
+
+### Smart Contract Creation and Execution Transactions
+
+The `relay` will **only** be charged if handling of the `EthereumTransaction` fails **before** any bytecodes are executed.
+In other words, the `relay` is charged when one of the pre-requisite checks for `EthereumTransaction` fails, regardless of
+the execution status of the transaction itself once it begins smart contract creation or execution in the EVM.
+
+Examples:
+ - The `relay` is charged if the `evm tx` cannot be parsed or is otherwise malformed
+ - The `relay` is charged if the `EOA` specified in the `evm tx` refers to a non-existent account
+ - The `relay` is charged if the `evm tx` was not properly signed by the keys of the `EOA`
+ - The `relay` is charged if the `EOA` has insufficient funds to pay the intrinsic gas fee
+ - The `relay` is charged if the `nonce` check fails
+ - The `relay` is not charged if the contract is called but concludes with an error code 
+ - The `relay` is not charged if the contract is called and concludes without error
+
+In summary the `relay` account is not charged if the `EthereumTransaction` transaction completes in a 
+`CONTRACT_REVERT_EXECUTED` or `SUCCESS` HAPI response code, all other responses codes will result in a charge to the
+relay account.
+
+ ### Regular Transactions
+
+A "regular transaction" is a simple HBAR crypto transfer between two accounts. The `relay` will be charged for **any**
+failure of the transaction. This is required to maintain compatibility with the semantics in Ethereum, where a regular
+transaction will not be included in a block or executed if it will not succeed.
+
+Example:
+- The `relay` is charged if the `evm tx` cannot be parsed or is otherwise malformed
+- The `relay` is charged if the `EOA` specified in the `evm tx` refers to a non-existent account
+- The `relay` is charged if the `evm tx` was not properly signed by the keys of the `EOA`
+- The `relay` is charged if the `EOA` has insufficient funds to pay the intrinsic gas fee **and** the transfer
+- The `relay` is charged if the receiver has `receiverSigRequired` enabled
+- The `relay` is charged if the receiver doesn't exist, must be auto-created, and the `EOA` has insufficient funds to
+    pay for the account creation
+- The `relay` is charged if the `nonce` check fails
+
+In summary the `relay` account is not charged if the `EthereumTransaction` transaction completes in a `SUCCESS`
+HAPI response code, all other response codes will result in a charge to the `relay` account.
+
+## Backwards Compatibility
+
+No changes to transaction execution or record stream externalization is made in this HIP.
```

Comment:
As far as I understood the idea, if the transaction is correct the relay won't be the payer of the transaction anymore. This would mean the signer (the `from` fieldn) and the payer will be the same account. Even if the record stream or proto structures are not changed we should make sure that eventually, the `signer=payer` part will be properly reflected in the implementation and the externalisation of the record.

Reply:
Hi @IvanKavaldzhiev I suggest not mixing the signer (from) and the payer concepts, regardless who is charged for the fees. The wrapped ethereum tx remains signed by the sender, and the "envelop" transaction still contains the payer's signature. Both accounts can be charged based on the outcome of the tx. Even if the payer is charged 0, it is still a potential payer, it set a maximum amount of hbar it allows to be spent, etc. This is the same concept now where, in case of some errors, only the payer is charged and not the sender (i.e. WRONG_NONCE: https://hashscan.io/testnet/transaction/1730913223.447111312)

Reply:
Your idea is keeping the payer concept to reflect the relayer account even if we don't charge it in case of successful transaction? You have a point that if signer=payer, then we will miss track of which relayer account has sent the transaction to the consensus node.

Reply:
@Neurone is correct.
No change to that logic, separate concepts.

HAPI sender/payer and the internal tx sender are by definition always different in the relay case when using EVM tools.
The only time they'd be the same is if the EOA was running their own relay using their account as the relay operator and they submitted an Eth TX to themselves.m or if they used the SDK directly to submit an `EthereumTransaction` in the same manner as the relay

Reply:
I see, I agree with your point. It was an important aspect to clear out in this discussion.